ssl/ssl_lib.c

```diff
@@ -3826,8 +3826,8 @@ SSL_CTX *SSL_CTX_new_ex(OSSL_LIB_CTX *libctx, const char *propq,
     }
 
     ret->method = meth;
-    ret->min_proto_version = 0;
-    ret->max_proto_version = 0;
+    ret->min_proto_version = meth->version == OSSL_QUIC_ANY_VERSION ? TLS1_3_VERSION : 0;
+    ret->max_proto_version = meth->version == OSSL_QUIC_ANY_VERSION ? TLS1_3_VERSION : 0;
```

Comment:
I don't think this is correct. "min_proto_version"/"max_proto_version" need to do something sensible when provided with *QUIC* version numbers not TLS ones, e.g. `SSL_set_min_proto_version(s, OSSL_QUIC1_VERSION)` needs to do the right thing and work.

Reply:
> I don't think this is correct. "min_proto_version"/"max_proto_version" need to do something sensible when provided with _QUIC_ version numbers not TLS ones, e.g. `SSL_set_min_proto_version(s, OSSL_QUIC1_VERSION)` needs to do the right thing and work.

The set_min_proto/set_max_proto calls with OSSL_QUIC1_VERSION should be just ignored.
Internally we use the min_proto/max_proto version for the inner TLS object and I do not think making this internal implementation future-proof for future QUIC versions at this point is necessary. It can be always changed regardless.

Reply:
There is one more problem with this SSL_set_min_proto_version() API in regards to QUIC versions - in theory the version numbers for future QUIC versions could conflict with TLS versions and then we might have a problem to recognize whether a QUIC or TLS version was meant by the caller.

Reply:
This still doesn't look right to me. This means that `SSL_CTX_get_min_proto_version` and `SSL_CTX_get_max_proto_version` will return the wrong thing for QUIC based SSL_CTX objects.

Why is this change even necessary if you keep the original call to `SSL_set_min_proto_version()` in quic_tls.c?

Reply:
I can drop this change, however it won't fix the SSL_CTX_get_min/max_proto_version() anyway. It is also very much questionable what these calls should return for QUIC.

Reply:
It seems quite clear to me. They should return 0 if no explicit min/max has been set, or OSSL_QUIC1_VERSION otherwise.

Reply:
In that case even with the current implementation they won't behave like this.

Reply:
Right...that's another PR

Reply:
#20790